File: persons-finder-projects/java/persons-finder-4j/src/main/java/com/persons/finder/data/Location.java
/**
 * 
 */
package com.persons.finder.data;

import javax.persistence.Entity;

import lombok.Getter;
import lombok.Setter;

/**
 * 
 */
@Entity
public class Location extends IndexField{

//	@Id
//    @GeneratedValue(strategy = GenerationType.IDENTITY)
//	@Getter @Setter private Long id;
	@Getter @Setter private Long referenceId;
	@Getter @Setter private Double latitude;
	@Getter @Setter private Double longitude;
}

File: persons-finder-projects/kotlin/persons-finder/src/main/kotlin/com/persons/finder/data/Location.kt
package com.persons.finder.data

import javax.persistence.Entity

@Entity
data class Location(
    // Tip: Person's id can be used for this field
    var referenceId: Long,
    var latitude: Double,
    var longitude: Double

) : IndexField() {
    constructor() : this(0L, 0.0, 0.0) {
    }
}
